OneCommunityGlobal/HighestGoodNetworkApp
# Vishala phase2 log materials bm dashboard

# Description
This feature is a part of Building Management Dashboard (bmdashboard). This feature is to log the materials used by the housing projects.
Implements # Phase 2 WBS : 3.3.3
Documents and Figma wireframes referred : https://docs.google.com/document/d/1WONWQW_56pXlyZpdQFibddFuT3iMxFjho-wfGwbHFV8/edit
https://www.figma.com/file/kpwXZP2pLtOJ1Fn2u7Sivn/HGN-PII?type=design&node-id=430-5449&mode=design&t=Br43OtYXP9erhdFF-0

## Related PRS (if any):
This frontend PR is related to the backend PR#583 (Vishala phase2 logmaterials)  (https://github.com/OneCommunityGlobal/HGNRest/pull/583)

## Main changes explained:
- Created new files to maintain the log materials feature
- Added urls , actions , constants , reducers to integrate with the log materials components

## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. on the related backend pr , do `npm run build` and `npm start`
4. log as admin user
5. go to http://localhost:3000/bmdashboard/logMaterials
6. The date is set to today's date initially
7. Select a project to load materials (At the moment , Tim and Vishala projects has some test data)
8. The projects loaded belong to Housing category only
9. If one user selects “project 1” and “Check Out”, this table will only show the rows representing the materials that belong to project 1 and the Available value is larger than 0. If the user selects “Check In”, the rows rendered from the database will be the materials whose Used value is larger than 0. 
10. Validations - Date : Date is less than or equal to current date. 
11. Validations - Log Entry : If Check In is selected , the entered value must be less than or equal to Using and if Check out is selected , the entered value must be less than or equal to Available
12. Cancel button : Clears the entered log values
13. Submit button : Posts the log entries to database.
14. Please test for the available functionalities. As it is a phase 2 task , I have implemented this with some inputs from the docs and based on my own ideas. So , please test this completely on multiple aspects.


## Screenshots or videos of changes:

<img width="858" alt="image" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/65274029/73e3fecf-ae7e-4c24-a6b2-f5bb75d3f68a">



[Untitled_ Oct 24,2023 10_27 AM.webm](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/65274029/ba522212-51a6-4886-9c6d-a1587d8d3780)
